### PR TITLE
fix Max-Age handling with tests

### LINKED ISSUE #6: handling of expired cookies
I make frequent use of HTTP::Tiny with HTTP::CookieJar, and today ran into a quirk in the way it handles expired cookies that may be considered a bug.

When a site returns a `Set-Cookie` header with an expiration date in the past, or with `Max-Age` <= 0, that cookie is supposed to be deleted. However, HTTP::CookieJar does not seem to handle this correctly. Instead, it first sets the expiration date to the current time + `Max-Age` and then checks if the expiration date is less than the current time, deleting the cookie if true. However, since this is an integer comparison, and since it happens quickly, cookies with `Max-Age` == 0 are not deleted.

Then, if another HTTP::Tiny request is performed within the same epoch second, incorrect cookies can be sent.

I believe the following change would conform to specifications:

From
```perl
    # if cookie has expired, purge any old matching cookie, too
    if ( defined $parse->{expires} && $parse->{expires} < $now ) {
        delete $self->{store}{$domain}{$path}{$name};
    }
```
to
```perl
    # if cookie has expired, purge any old matching cookie, too
    if (
        (defined $parse->{expires} && $parse->{expires} < $now)
     || (defined $parse->{'max-age'} && $parse->{'max-age'} <= 0)
    ) {
        delete $self->{store}{$domain}{$path}{$name};
    }
```

If you agree with my reasoning on this, I would be happy to submit a PR.

### PR DESCRIPTION
**Pull request by jvolkening**
Fixes #6 

Not sure how/if to manually edit Changes or if dzil does that automatically.

### CODE REVIEW

File: t/add.t [hunk 1]
```diff
@@ -207,6 +207,33 @@ my @cases = (
...
     },
+    # check that Max-Age supercedes Expires and that Max-Age <= 0 forces
+    # expiration
+    {
+        label   => "max-age supercedes expires",
+        request => "http://example.com/",
+        cookies => [
+            "lang=en-us; Max-Age=100; Expires=Thu, 1 Jan 1970 00:00:00 GMT",
+            "SID=31d4d96e407aad42; Expires=Thu, 3 Jan 4841 00:00:00 GMT",
+            "SID=0000000000000000; Max-Age=0",
```
Comment (xdg) [inline t/add.t:218]: Could you please add a valid expires here (maybe after Max-Age)?  I'd like to verify that max-age supersedes an expires in the same cookie.

**Review by xdg — CHANGES_REQUESTED** [review]
Looks good.  I'd like the test to be a little more robust.  For Changes, just add it after `{{$Next}}`:

```
{{$NEXT}}

<your change here>

0.008     2015-11-06 22:48:19-05:00 America/New_York
```
And again, thank you!

File: t/add.t (hunk 1 above)
Comment (jvolkening) [reply t/add.t]: Latest push now tests three conditions:

1. `Max-Age` in future, `Expires` in past

2. `Max-Age` in past, `Expires` in future

3. `Max-Age` == 0, `Expires` in future

Only the first cookie should be saved. Happy to add any additional tests...I just can't see what they should be.

**Comment by xdg** [comment]
Thanks!  I'll ship a dev release to CPAN shortly and let it smoke a couple days before I ship a final.

**Pull request merged**